### PR TITLE
Maze paint setup

### PR DESCRIPTION
**Pull request by marijnvdwerf**
Branched from the spiral slide branch because of method comments

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
For reference:

The bit for each wall:
image

The different segments used:
image

**Comment by IntelOrca** [comment]
I like these diagrams you keep making. Maybe we could ASCII-fy them and put them in the source code. :smile:

File: src/ride/track_paint.c [hunk 1]
```diff
@@ -982,6 +982,278 @@ TRACK_PAINT_FUNCTION get_track_paint_function_topspin(int trackType, int directi
...
+	}
+
+
+	if (maze_entry & (1 << 2 | 1 << 6 | 1 << 10 | 1 << 14)) {
+		RCT2_GLOBAL(0x009DEA52, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA54, uint16) = 15;
+		RCT2_GLOBAL(0x009DEA56, sint16) = height + 2;
+		sub_98197C(image_id + SPR_MAZE_OFFSET_COLUMN_CENTER, 14, 14, 2, 2, 8, height, get_current_rotation());
+
+		RCT2_GLOBAL(0x141E9C4, uint16) = height + 12;
```
Comment (marijnvdwerf) [inline src/ride/track_paint.c]: @duncanspumpkin Can you check this (and the following) assignment? I don't think this is as it should be.

**Comment by marijnvdwerf** [comment]
@IntelOrca @duncanspumpkin This one should be ready for testing + merging. Only changes I can imagine would be file-splitting or addition of a flag-setting utility.

**Comment by duncanspumpkin** [comment]
Okay I've checked this and it looks all fine.

**Comment by janisozaur** [comment]
Tested by building few mazes. Looks good.

**Pull request merged**